### ISSUE TITLE
[Feature Request] Disable Kodi Controller input during launch

### ISSUE THREAD

**Issue by zach-morris** [chrisism/script.module.ael#7]
I think this is the right repo, but forgive me if it isn't.

Feature request was originally made for AEL [here](https://github.com/Wintermute0110/plugin.program.AEL.dev/issues/151).

Issue is that sometimes Kodi doesn't seem to be able to understand if it's not the active / front window, resulting in controller input for the launched process being picked up by Kodi as well.  I think the issue is caused by how the window for the launched game is created.

The request is to workaround that by adding a launch option to disable Kodi input prior to launch, then re-enabling it after returning to Kodi.

Note that I already do this in IAGL, and it works just fine. You can disable Kodi controller input with:
```
xbmc.executeJSONRPC('{"jsonrpc":"2.0","method":"Settings.SetSettingValue","params":{"setting":"input.enablejoystick","value":false},"id":"1"}')
```

And then re-enable it after return to kodi with:
```
xbmc.executeJSONRPC('{"jsonrpc":"2.0","method":"Settings.SetSettingValue","params":{"setting":"input.enablejoystick","value":true},"id":"1"}')
```

**Comment by chrisism** [comment]
Added feature to the backlog (https://dev.azure.com/jnpro/AKL/_workitems/recentlyupdated).
Item #14 (https://dev.azure.com/jnpro/AKL/_workitems/edit/14/). Will pick it up after current gamestream addon fixes are done.

It is the correct repository and original code can be found in the lib/akl/launchers.py file. Of course PRs are welcome with the changes before I will pick it up myself. In that case please gitflow strategy and create a feature/<name> branch based on dev and submit that branch for the PR.

**Comment by chrisism** [comment]
@zach-morris Feature has been rolled out in script.module.akl v1.0.5.
You can now activate this trigger by setting the corresponding flag in the ExecutionSettings object, which you give to the constructor of the Launcher implementation. 
Default value is False, so not enabled. Currently busy adding this feature in the default launcher etc. but not yet planned a release there. So if you are waiting on this feature to be applied in script.akl.default or script.akl.retroarchlauncher, let me know.
However, if you are building custom launchers you can already use it of course.

Example getting the setting from settings.xml and injecting it in the launcher:
`python
        execution_settings.suspend_joystick_engine  = settings.getSettingAsBool('suspend_joystick')
        auncher = AppLauncher(
            args.akl_addon_id, 
            ....
            executor_factory, 
            execution_settings)
`

**Comment by chrisism** [comment]
#8 Fixed in v1.0.5

**Comment by chrisism** [comment]
#8 Fixed in v1.0.5

**Comment by zach-morris** [comment]
> @zach-morris Feature has been rolled out in script.module.akl v1.0.5. You can now activate this trigger by setting the corresponding flag in the ExecutionSettings object, which you give to the constructor of the Launcher implementation. Default value is False, so not enabled. Currently busy adding this feature in the default launcher etc. but not yet planned a release there. So if you are waiting on this feature to be applied in script.akl.default or script.akl.retroarchlauncher, let me know. However, if you are building custom launchers you can already use it of course.
> 
> Example getting the setting from settings.xml and injecting it in the launcher: `python execution_settings.suspend_joystick_engine = settings.getSettingAsBool('suspend_joystick') auncher = AppLauncher( args.akl_addon_id, .... executor_factory, execution_settings) `

Awesome.  I'll give it a try.  Thanks very much
